### PR TITLE
fix: ctrl+c on examples copies the whole text

### LINKED ISSUE #53: ctrl+c on examples copies the whole text, not just the selection, that's disturbing

### PR DESCRIPTION
**Pull request by AkshataABhat**
closes #53

### CODE REVIEW

**Comment by JeelRajodiya** [comment]
**Great work** @AkshataABhat!  We have solve the problem, but I noticed one side effect of these changes. 
Some of the codeblocks have few lines highlighted in green, and since we are disabling the line wrapping. there will be no elements that wrap the lines, so the green highlights have disappeared. Can you please also resolve it?

image

**Review by JeelRajodiya — CHANGES_REQUESTED** [review]
**Great work** @AkshataABhat!  We have solve the problem, but I noticed one side effect of these changes. 
Some of the codeblocks have few lines highlighted in green, and since we are disabling the line wrapping. there will be no elements that wrap the lines, so the green highlights have disappeared. Can you please also resolve it?

image

**Comment by AkshataABhat** [comment]
Yes, will do @JeelRajodiya 👍

**Comment by JeelRajodiya** [comment]
Any updates @AkshataABhat?

**Comment by AkshataABhat** [comment]
@JeelRajodiya pls check this pr.

**Review by JeelRajodiya — CHANGES_REQUESTED** [review]
Can you please ensure that these highlights have the full width (same as the codeblock) and does not have gaps between the lines? 

Screenshot 2024-09-22 134033

**Comment by AkshataABhat** [comment]
Done, please check.

File: app/components/CodeSnippet/CodeSnippet.tsx [hunk 1]
```diff
@@ -71,9 +63,10 @@ export default function CodeSnippet({
         customStyle={{
           paddingInline: 0,
           paddingBlock: "10px",
+          whiteSpace: "pre-wrap",
```
Comment (JeelRajodiya) [inline app/components/CodeSnippet/CodeSnippet.tsx:66]: please remove this style.
```suggestion
```

File: app/components/CodeSnippet/CodeSnippet.tsx [hunk 2]
```diff
@@ -82,10 +75,13 @@ export default function CodeSnippet({
         startingLineNumber={startingLineNumber}
         lineProps={(lineNumber) => {
           let style: CSSProperties = {
-            opacity: colorMode === "dark" ? 0.9 : 1,
+            display: "block",
+            width: "100%",
```
Comment (JeelRajodiya) [inline app/components/CodeSnippet/CodeSnippet.tsx:79]: Please remove this style as well, the blocks seem to have 100% width by default. 
```suggestion
```

**Review by JeelRajodiya — CHANGES_REQUESTED** [review]

File: app/components/CodeSnippet/CodeSnippet.tsx (hunk 2 above)
Comment (JeelRajodiya) [inline app/components/CodeSnippet/CodeSnippet.tsx:79]: Please remove this style as well, the blocks seem to have 100% width by default.
```suggestion
```

**Comment by JeelRajodiya** [comment]
Congrats Akshata! it works perfectly!

**Pull request merged**